doc/jenkins.rst

```diff
@@ -0,0 +1,136 @@
+Integration with Jenkins
+========================
+
+`Universum` requires no special integration with `Jenkins`, except for one thing. It is usually launched
+as one long step in a single build stage. Because of that, the whole `Universum` log is printed as a plain text
+without navigation anchors.
+
+Compare with these:
+
+* When ran locally, all Universum build step logs are stored to separate files instead of printing their
+  output to console
+* When launched by TeamCity, `Unviersum` uses `service messages
+  <https://www.jetbrains.com/help/teamcity/service-messages.html>`__ to increase log readability
+
+To simplify navigating long logs and finding relevant information on build results, we provide a user-friendly
+interactive log with collapsible blocks and other features.
+
+.. warning::
+
+    By default, Jenkins `does not render interactive content <https://www.jenkins.io/doc/book/security/user-content/>`__.
+    This means that without changing server settings, interactive features of the generated log will be
+    inaccessible when opened directly from Jenkins artifacts.
+
+
+Here's the list of steps we performed to integrate the interactive log with Jenkins server:
+
+1. :ref:`Add a command line option <jenkins#add_arg>` to generate a self-contained HTML file
+2. :ref:`Add a Jenkins plugin <jenkins#plugin>` to integrate a generated log into a Jenkins job
+3. :ref:`Set up Resource Root URL <jenkins#resource_url>` to allow Jenkins rendering interactive content
+4. :ref:`Configure reverse proxy <jenkins#nginx>` to handle multiple domains interaction
+
+
+.. _jenkins#add_arg:
+
+Add command line option
+-----------------------
+
+To generate a single interactive self-contained HTML file, pass an `--html-log <args.html#Output>`__
+option to command line. It will be stored to in project `artifacts <args.html#Artifact\ collection>`__ folder.
+Note that the log name can either be specified or left default.
+
+.. note::
+
+    Jenkins jobs do not show any files via web interface if not not specifically configured to do so. Use
+    ``archiveArtifacts`` in Pipeline or ``Archive the artifacts`` in `Post-build Actions` to check the file presence
+    and contents before next step if needed
+
+
+.. _jenkins#plugin:
+
+Add a Jenkins plugin
+--------------------
+
+A `HTML Publisher <https://plugins.jenkins.io/htmlpublisher/>`__ plugin is a very convenient way to add an HTML report
+to a Jenkins job. To use it, you will need to:
+
+1. Install it on server (server restart might be needed to apply changes)
+2. Add ``-hl``/``--html-log``/``-hl <name>`` option to the `Universum` command line as mentioned in previous section
```

Comment:
I think you can remove this item if it is already mentioned in the previous section